src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
+}
 
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue'
+}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = Shape.triangle;
 
-}
+  getArea(): number {
+    const s = 0.5 * (this.a + this.b + this.c);
+    const A = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
-export class Circle {
+    return Math.floor(A * 100) / 100;
+  }
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const arraySides = [a, b, c];
+    const positiveFilter = arraySides.filter((num) => num <= 0);
+    const sides = arraySides.sort((current, next) => next - current);
+
+    if (positiveFilter.length > 0 || sides[0] >= sides[1] + sides[2]) {
+      throw new Error('can`t formed triangle');
+    }
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = Shape.circle;
+
+  getArea():number {
+    return Math.floor(this.radius ** 2 * Math.PI * 100) / 100;
+  }
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('radius must be positive number');
+    }
```

Comment:
you have to check if all arguments are positive in each constructor, so would be better to use a function for this and avoid duplicating